File: day03/d3p1.ml
let score = 0
let characters = Hashtbl.create 1000
  
let base = Char.code 'A' - 27;;
for i = 27 to 52 do
  let c = Char.chr (i + base) in
  Hashtbl.add characters c i
done

let base = Char.code 'a' - 1;;
for i = 1 to 26 do
  let c = Char.chr (i + base) in
  Hashtbl.add characters c i
done
  





let file = open_in "input.txt";;
try
  while true do
    let 
      line = input_line file
    in
      {
      for c = 0 to ((String.length line)/2 - 1) do
        let bag_one = Hashtbl.create 1000
        Hashtbl.add bag_one line.[c] 1
      done
      for c = ((String.length line)/2) to (String.length line - 1) do
        if Hashtbl.mem bag_one line.[c] then
          score = score + (Hashtbl.find characters line.[c])
      done
      }
  done
with End_of_file ->
  close_in file

print_endline (string_of_int score)

File: day03/d3p1.py

score = {chr(i+96): i for i in range(1,27)}
capitals = {chr(i+38): i for i in range(27,53)}
score.update(capitals)

count = 0
with open('input.txt', 'r') as f:
    for l in f:
        line = l.strip()
        checker = set()
        n = len(line) // 2
        for i in range(n):
            checker.add(line[i])
        for i in range(n):
            if line[-i-1] in checker:
                count += score[line[-i-1]]
                break

print(count)


